common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -126,4 +150,50 @@ private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
       }
     }
   }
+
+  private class DownloadCallback implements StreamCallback {
+
+    private WritableByteChannel channel = null;
+    private File targetFile = null;
+    private int chunkIndex;
+
+    public DownloadCallback(File targetFile, int chunkIndex) throws IOException {
+      this.targetFile = targetFile;
+      this.channel = Channels.newChannel(new FileOutputStream(targetFile));
+      this.chunkIndex = chunkIndex;
+    }
+
+    @Override
+    public void onData(String streamId, ByteBuffer buf) throws IOException {
+      channel.write(buf);
+    }
+
+    @Override
+    public void onComplete(String streamId) throws IOException {
+      channel.close();
+      ManagedBuffer buffer = new FileSegmentManagedBuffer(
+        transportConf, targetFile, 0, targetFile.length()) {
+        @Override
+        public ManagedBuffer release() {
+          ManagedBuffer ret = super.release();
+          if (!targetFile.delete()) {
+            logger.warn("Failed to cleanup " + targetFile.getAbsolutePath());
```

Comment:
Perhaps WARN is too strong here, INFO or lower should do.
It is possible for files to be cleaned up as part of shutdown (for example) and we dont want lot of warn messages in log